### PR TITLE
Filter customizers only for channels that support customization

### LINKED ISSUE #804: Expanding detail channel shows empty popup
- Better add a customizer for text's `format`
- Better not show caret for detail and make it unclickable

### PR DESCRIPTION
**Pull request by ssharif6**
Fix #804 

Disable customizer popup caret option for Detail channel. 
Created constant for channels that support encoding customization.

Brief demo showing customizer doesn't display caret for channels that have no customization supported yet. 

detailchannel

### CODE REVIEW

File: src/components/encoding-pane/encoding-shelf.tsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ import {DraggedFieldIdentifier, Field} from '../field/index';
 import * as styles from './encoding-shelf.scss';
 import {FieldCustomizer} from './field-customizer';
 import {FunctionPicker, FunctionPickerWildcardHandler} from './function-picker';
+import { CUSTOMIZABLE_ENCODING_CHANNELS } from './property-editor-schema';
```
Comment (kanitw) [inline src/components/encoding-pane/encoding-shelf.tsx]: Please do not include space between braces.  If we don't have this a part of linter yet, please submit a separate PR to enforce this.

File: src/components/encoding-pane/property-editor-schema.ts [hunk 2]
```diff
@@ -58,6 +58,9 @@ export interface PropertyEditorSchema {
   schema: ObjectSchema;
 }
 
+// Currently supported customizble encoding channels that display caret in customizer UI
+export const CUSTOMIZABLE_ENCODING_CHANNELS = [Channel.X, Channel.Y, Channel.COLOR, Channel.SIZE, Channel.SHAPE];
```
Comment (kanitw) [inline src/components/encoding-pane/property-editor-schema.ts]: How about row and column?

File: src/components/encoding-pane/encoding-shelf.tsx (hunk 1 above)
Comment (ssharif6) [reply src/components/encoding-pane/encoding-shelf.tsx]: Sorry every time I use VS Code this happens. Will make a PR to add this rule to linter.

File: src/components/encoding-pane/property-editor-schema.ts (hunk 2 above)
Comment (ssharif6) [reply src/components/encoding-pane/property-editor-schema.ts]: Those are not supported as of now. Eventually it'd be nice to customize the row/column encoding channels. It'd be nice to support customizing headers and whatnot. 

So better just leave what is working right now, and then when add support for row/column, update this list as well
